src/thermo/Phase.cpp

```diff
@@ -846,6 +846,7 @@ bool Phase::addSpecies(shared_ptr<Species> spec) {
     wt = std::max(wt, Tiny);
     m_molwts.push_back(wt);
     m_rmolwts.push_back(1.0/wt);
+    spec->setMolecularWeight(wt);
```

Comment:
Given that this can throw, I think we should move the calculation of `wt` and this call to before any of the arrays belonging to the phase have values appended. I think this starts where `m_speciesNames` is updated. Otherwise, this will result in the `Phase` object ending up in an inconsistent state.